apps/aens/src/aens_hash.erl

```diff
@@ -28,14 +34,37 @@
 %%% API
 %%%===================================================================
 
+
 -spec commitment_hash(binary(), integer()) -> commitment_hash().
 commitment_hash(NameAscii, Salt) ->
+    case aens_utils:name_domain(NameAscii) of
+        {ok, Domain} when Domain =:= <<"aet">> ->
```

Comment:
This pulling of domain (split binary, reverse list, check first elem) seems rather inefficient, but it's probably not an issue in reality.

Could be made faster and cleaner though:

```
is_registrar(<<Name/binary>>, <<Registrar/binary>>) when size(Name) > size(Registrar) ->
    Reg = <<".", Registrar/binary>>,
    RegSize = size(Reg),
    binary:part(Name, size(Name) - RegSize, RegSize) == Reg;
is_registrar(_, _) ->
    false.
```

Reply:
No need for efficieny here... clear code prefered